graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/ProjectionSerializer.kt

```diff
@@ -19,16 +19,24 @@ package com.netflix.graphql.dgs.client.codegen
 import java.util.*
 
 class ProjectionSerializer(private val inputValueSerializer: InputValueSerializer) {
+
     fun serialize(projection: BaseProjectionNode, isFragment: Boolean = false): String {
         if (projection.fields.isEmpty() && projection.fragments.isEmpty()) {
             return ""
         }
 
         val prefix = if (isFragment) {
-            "... on ${projection::class.java.name.substringAfterLast(".").substringAfterLast("_").substringBefore("Projection")} { "
+            val schemaType = projection.schemaType.orElse(
```

Comment:
The name `schemaType` is a bit too generic. I don't immediately have a better alternative though, let's see if we can come up with something while talking this through, otherwise we keep this name.